make/Program.mk

```diff
@@ -69,30 +69,27 @@ endif
 
 .PHONY: flash
 flash: all test_softdevice flash_mbr
-	$(Q)printf "r\n" > $(BUILDDIR)flash.jlink
-ifdef ID
-	$(Q)printf "w4 $(ID_FLASH_LOCATION), 0x$(ID_SECON) 0x$(ID_FIRST)\n" >> $(BUILDDIR)flash.jlink
-endif
 ifeq ($(strip $(grep -q Microsoft /proc/version)),)
 	@echo Flashing: $(HEX)
 	nrfjprog.exe -f nrf52 --program $(HEX) --sectorerase
 	nrfjprog.exe -f nrf52 --reset
 else ifeq ($(USE_BOOTLOADER),1)
+ifdef ID
+	$(Q)printf "w4 $(ID_FLASH_LOCATION), 0x$(ID_SECON) 0x$(ID_FIRST)\n" >> $(BUILDDIR)flash.jlink
+endif
+	$(Q)printf "r\n" > $(BUILDDIR)flash.jlink
 	$(Q)printf "loadfile $(MERGED_HEX) \nr\ng\nexit\n" >> $(BUILDDIR)flash.jlink
 	$(Q)$(JLINK) $(JLINK_FLAGS) $(BUILDDIR)flash.jlink
 else ifeq ($(USE_BOOTLOADER),0)
+ifdef ID
+	$(Q)printf "w4 $(ID_FLASH_LOCATION), 0x$(ID_SECON) 0x$(ID_FIRST)\n" >> $(BUILDDIR)flash.jlink
+endif
+	$(Q)printf "r\n" > $(BUILDDIR)flash.jlink
```

Comment:
The "r\n" line MUST come before the `ifdef ID`

Reply:
Done. Thanks.